### PR TITLE
UI: Adds a button to clear all selected filters

### PR DESCRIPTION
**Pull request by pratap0007**
This patch adds following

- adds an action in resource store to clear all selected filters
- adds a button on `No Resource Found` page

Signed-off-by: Shiv Verma <shverma@redhat.com>

Screenshot from 2021-02-22 16-33-30

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/pipeline/blob/master/CONTRIBUTING.md) for more details._

### CODE REVIEW

File: ui/src/containers/Resources/index.tsx [hunk 1]
```diff
@@ -17,17 +18,20 @@ import './Resources.css';
 const Resources = () => {
   const { resources } = useMst();
 
-  const checkResources = (resources: IResource[]) => {
-    return !resources.length ? (
+  const checkResources = (resourceList: IResource[]) => {
```
Comment (PuneetPunamiya) [inline ui/src/containers/Resources/index.tsx:21]: Any reason why are we changing the variable name here ??

File: ui/src/containers/SortDropDown/index.tsx [hunk 2]
```diff
@@ -1,15 +1,14 @@
 import React, { useState } from 'react';
 import { Select, SelectVariant, SelectOption, SelectOptionObject } from '@patternfly/react-core';
-import { useObserver } from 'mobx-react';
+import { observer } from 'mobx-react';
 import { useMst } from '../../store/root';
 import { SortByFields } from '../../store/resource';
 import './SortDropDown.css';
 
-const Sort: React.FC = () => {
+const Sort: React.FC = observer(() => {
```
Comment (PuneetPunamiya) [inline ui/src/containers/SortDropDown/index.tsx:8]: Why this change ?? Is it not working if we use `useObserver` in return which renders as the state updates

File: ui/src/containers/Resources/index.tsx (hunk 1 above)
Comment (pratap0007) [reply ui/src/containers/Resources/index.tsx:21]: within function we needed `resources` store and function has local variable (function params) with same name ..

File: ui/src/containers/Resources/index.tsx [hunk 3]
```diff
@@ -17,17 +18,20 @@ import './Resources.css';
...
+    return !resourceList.length ? (
       <EmptyState variant={EmptyStateVariant.full} className="hub-resource-emptystate__margin">
         <EmptyStateIcon icon={CubesIcon} />
         <Title headingLevel="h5" size="md">
           No Resource Found.
         </Title>
+        <Button variant="primary" onClick={() => resources.clear()}>
+          Clear All Filters
```
Comment (PuneetPunamiya) [inline ui/src/containers/Resources/index.tsx:29]: What would it show on UI if there are no resources added in the store or let's say we are getting zero resources from api, will it still show `Clear All Filters`

File: ui/src/containers/Resources/index.tsx (hunk 1 above)
Comment (PuneetPunamiya) [reply ui/src/containers/Resources/index.tsx:21]: Should we just call it as `items`

File: ui/src/containers/Search/index.tsx [hunk 4]
```diff
@@ -23,9 +21,9 @@ const Search: React.FC = () => {
     return;
   };
 
-  return (
+  return useObserver(() => (
```
Comment (PuneetPunamiya) [inline ui/src/containers/Search/index.tsx:24]: Is this change related to this pr ??

File: ui/src/containers/Resources/index.tsx [hunk 5]
```diff
@@ -17,17 +18,20 @@ import './Resources.css';
 const Resources = () => {
   const { resources } = useMst();
 
-  const checkResources = (resources: IResource[]) => {
-    return !resources.length ? (
+  const checkResources = (resourceList: IResource[]) => {
+    return !resourceList.length ? (
```
Comment (PuneetPunamiya) [inline ui/src/containers/Resources/index.tsx:22]: Could you please consider writing a unit test for different conditions ??

File: ui/src/containers/Resources/index.tsx (hunk 1 above)
Comment (pratap0007) [reply ui/src/containers/Resources/index.tsx:21]: okay will change it

File: ui/src/containers/Resources/index.tsx (hunk 3 above)
Comment (pratap0007) [reply ui/src/containers/Resources/index.tsx:29]: nope

File: ui/src/containers/Resources/index.tsx (hunk 5 above)
Comment (pratap0007) [reply ui/src/containers/Resources/index.tsx:22]: I guess for that there is a test already

File: ui/src/containers/Search/index.tsx (hunk 4 above)
Comment (pratap0007) [reply ui/src/containers/Search/index.tsx:24]: yeah

File: ui/src/containers/Resources/index.tsx (hunk 3 above)
Comment (PuneetPunamiya) [reply ui/src/containers/Resources/index.tsx:29]: Then what it would show on UI ??

Comment (PuneetPunamiya) [reply ui/src/containers/Resources/index.tsx:29]: Just a spinner right ??

File: ui/src/containers/Resources/index.tsx (hunk 5 above)
Comment (PuneetPunamiya) [reply ui/src/containers/Resources/index.tsx:22]: I think there is test for empty state, I guess we need to write for `Clear All Filters` as well

File: ui/src/containers/Resources/index.tsx (hunk 3 above)
Comment (pratap0007) [reply ui/src/containers/Resources/index.tsx:29]: yeah, for that condition we will need to handle it by `resource` api response error

**Review by vinamra28 — APPROVED** [review]
/lgtm
